packages/provider/src/transports/wallet-request-handler.ts

```diff
@@ -252,43 +249,66 @@ export class WalletRequestHandler implements ExternalProvider, JsonRpcHandler, P
           break
         }
 
-        case 'personal_sign': {
+        case 'personal_sign':
+        case 'eth_sign': {
           // note: message from json-rpc input is in hex format
-          const [message, signingAddress] = request.params!
-
-          let sig = ''
-          // TODO:
-          // if (process.env.TEST_MODE === 'true' && this.prompter === null) {
-          if (this.prompter === null) {
-            // prompter is null, so we'll sign from here
-            sig = await signer.signMessage(ethers.utils.arrayify(message), chainId)
-          } else {
-            // prompt user to provide the response
-            sig = await this.prompter.promptSignMessage({ chainId: chainId, message: message })
-          }
-
-          if (sig && sig.length > 0) {
-            response.result = sig
+          let message: any
+
+          // there is a difference in the order of the params:
+          // personal_sign: [data, address]
+          // eth_sign: [address, data]
+          if (request.method === 'personal_sign') {
+            const [data, address] = request.params!
+            message = data
           } else {
-            // The user has declined the request when value is null
-            throw new Error('declined by user')
+            const [address, data] = request.params!
+            message = data
           }
-          break
-        }
-
-        case 'eth_sign': {
-          // note: message from json-rpc input is in hex format
-          const [signingAddress, message] = request.params!
 
           let sig = ''
+
           // TODO:
           // if (process.env.TEST_MODE === 'true' && this.prompter === null) {
           if (this.prompter === null) {
             // prompter is null, so we'll sign from here
             sig = await signer.signMessage(ethers.utils.arrayify(message), chainId)
           } else {
-            // prompt user to provide the response
-            sig = await this.prompter.promptSignMessage({ chainId: chainId, message: message })
+            // check if wallet is deployed and up to date, if not, prompt user to deploy
+            let isDeployed: boolean
```

Comment:
Can we move this block of code to a standalone function so we don't have it duplicated on `eth_sign` and `eth_signTypedData`? this method could just return `isUpToDate` and assume that the wallet not being deployed is just another way of not being up to date